File: CPP_programming/Day7_inheritance-1/Employee/Person.cpp
 #include"Person.h"

Person::Person()////////////////////////constructors
{

}
Person::Person(Cstring nm, Date bd):name(nm),b_date(bd)
{
    //Cstring(nm);
    //Date(bd);
}
Person :: Person(const char * nm, int d, int m, int y):name(nm),b_date(d,m,y)
{
    //Cstring(nm);s
    //Date
}
Person::Person(Person &obj):name(obj.name), b_date(obj.b_date)
{
    // name = obj.name;
    // b_date = obj.b_date;
}
void Person::display()/////////////////////////facilitators
{
    name.display();
    b_date.display();
}

File: Core_Java/Day3/A8_Person/Person.java
class Person
{
	private String name;
	private Date bdate;
	
	public Person()
	{
		//name="Empty"; //or
		name= new String("Empty");
		bdate=new Date();//date obj is created on heap and default constructor is called
	}
	public Person(String name, int dd, int mm, int yy)
	{
		this.name=name;
		this.bdate=new Date(dd, mm, yy);
	}
	
	public void display()
	{
		System.out.println("Name: "+name);
		//System.out.println("bdate: "+bdate);
		bdate.showDate();
	}
	
}